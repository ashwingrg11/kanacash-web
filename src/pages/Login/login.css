.forget-signup-button{
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
}

.signup-button-login{
    border: none;
    background-color: transparent;
    color:#1077BE ;
    font-weight: bold;
}
.signup-button-login:hover{
    color:#0e3c65 ;
}
.forgetpassword-button{
    color: #828282;
}
.forgetpassword-button:hover{
    color: #6c757d;
}