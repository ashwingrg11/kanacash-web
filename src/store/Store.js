import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
