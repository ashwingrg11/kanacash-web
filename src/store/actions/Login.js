import {
  LOGIN_POST_API_REQUEST,
  LOGIN_POST_API_SUCCESS,
  LOGIN_POST_API_FAILURE,
} from "./constant/ActionTypes";

// import AsyncStorage from '@react-native-community/async-storage';
import * as api from "../../services/axios/authApi";

import { setError } from "./Error";
import { setToken } from "./Auth";
// import { showLoader, hideLoader } from "./Loader";

export const loginApiRequestAction = (loginRequest) => (dispatch, getState) => {
  dispatch({
    type: LOGIN_POST_API_REQUEST,
  });
  // dispatch(showLoader());
};

export const loginApiSuccessAction = (loginSuccess) => (dispatch, getState) => {
  dispatch({
    type: LOGIN_POST_API_SUCCESS,
    payload: loginSuccess,
  });
  dispatch(
    setToken({
      accessToken: loginSuccess.accessToken,
      tokenType: loginSuccess.tokenType,
    })
  );
};

export const loginApiFailureAction = (loginFailure) => (dispatch, getState) => {
  dispatch({
    type: LOGIN_POST_API_FAILURE,
    payload: loginFailure,
  });

  if (loginFailure !== undefined) {
    let modalConfig = {
      message: loginFailure.data.message,
      message_title: "Sorry!",
    };

    // dispatch(hideLoader());
    dispatch(setError(modalConfig));
  }
};

export const getWidgetToken = () => (dispatch, getState) => {
  api
    .getWidgetToken()
    .then((res) => {})
    .catch((err) => {
      // console.log(err);
    });
};

export const loginAction = (body) => (dispatch, getState) => {
  // dispatch(loginApiRequestAction());
  return new Promise((resolve, reject) => {
    api
      .login(body)
      .then((response) => {
        try {
          if (response.status === 200) {
            localStorage.setItem("accessToken", response.data.data.accessToken);
            localStorage.setItem("type", response.data.data.tokenType);
            localStorage.setItem("senderId", response.data.data.senderId);
            localStorage.setItem("userId", response.data.data.userId);
            localStorage.setItem("role", response.data.data.role);

            localStorage.setItem(
              "refreshToken",
              response.data.data.refreshToken
            );
            dispatch(loginApiSuccessAction(response.data.data));
            let responsed = {
              status: "success",
              role: response.data.data.role,
            };
            resolve(responsed);
          } else {
            let res = {
              status: "error",
              errorMessage: response.data.message,
            };
            reject(res);
          }
        } catch (error) {
          let response = {
            status: "error",
            errorMessage: error.response.data.message,
          };
          reject(response);
        }
      })
      .catch((error) => {
        dispatch(loginApiFailureAction(error));
        let response = {
          status: "error",
          errorMessage: error.data.message,
        };
        reject(response);
      });
  });
};

export const testLogin = () => (dispatch) => dispatch({ type: "TEST_LOGIN" });
