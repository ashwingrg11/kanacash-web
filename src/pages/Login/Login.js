import React, { Component } from "react";
import { Row, Col, Form, Modal } from "react-bootstrap";
import { Formik } from "formik";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import * as yup from "yup";
import PropTypes from "prop-types";
import ROUTES from "../../assets/Routes/Routes";
import ErrorCircle from "../../assets/images/error.svg";
import CrossButton from "../../assets/images/Cross.svg";
import { getCurrentUser } from "../../store/actions/UserDetailsActions";
import { loginAction } from "../../store/actions/Login";
import ButtonSpinner from "../../components/ButtonSpinner/ButtonSpinner";
import { getActiveAccountSideBar } from "../../store/actions/senderDetail";
import "./login.css";
class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      errors: "",
    };
  }

  onClickLogin = (values, { setSubmitting }) => {
    let body = {
      email: values.email, //Nepal12344@yopmail.com
      password: values.password, //Password!
      // device: "deviceID", //`${fingerPrint}`,
    };
    this.setState({ errors: "" });
    setSubmitting(true);
    this.props
      .loginAction(body)
      .then((res) => {
        setSubmitting(false);
        if (res.status === "success") {
          this.props.getCurrentUser();
          this.props.close();
          if (res.role === "USER") {
            this.props.history.push(`${ROUTES.WELCOME}`);
          } else {
            localStorage.setItem("activeSideBar", "AdminDashboard");
            this.props.getActiveAccountSideBar("AdminDashboard");
            this.props.history.push(`${ROUTES.ADMIN_DASHBOARD}`);
          }
        } else {
          this.setState({ errors: `${res.errorMessage}` });
        }
      })
      .catch((err) => {
        setSubmitting(false);
        this.setState({ errors: err.errorMessage });
        // console.log("err in login ", err);
      });
  };

  render() {
    let schema = yup.object({
      email: yup
        .string()
        .required("This field cannot be empty.")
        .email("Please enter a valid email address."),
      password: yup.string().required("This field cannot be empty."),
    });
    let initialValue = {
      email: "",
      password: "",
    };

    return (
      <Modal
        show={this.props.show}
        onHide={this.props.close}
        centered
        size="md"
      >
        <Modal.Body className="theme-modal">
          <div
            style={{
              position: "absolute",
              top: "-10px",
              right: "-10px",
              cursor: "pointer",
            }}
            onClick={() => {
              this.setState({ errors: "" });
              this.props.close();
            }}
          >
            <img src={CrossButton} alt="CrossButton" />
          </div>
          <h2 className="sub-title">Sign in</h2>
          {this.state.errors && (
            <Row>
              <Col
                style={{
                  color: "#dc3545",
                  backgroundColor: "rgba(255,0,0,.27)",
                  margin: "10px",
                  padding: "10px",
                  fontSize: "13px",
                }}
                className="d-flex justify-content-center"
              >
                {" "}
                <img
                  src={ErrorCircle}
                  alt="ErrorCircle"
                  height="10px"
                  width="10px"
                  style={{ margin: "6px 6px 0px 0px " }}
                />
                {this.state.errors}
              </Col>
            </Row>
          )}
          <Formik
            validationSchema={schema}
            onSubmit={this.onClickLogin}
            initialValues={initialValue}
          >
            {({
              handleSubmit,
              handleChange,
              handleBlur,
              values,
              touched,
              isValid,
              errors,
              isSubmitting,
            }) => (
              <Form
                noValidate
                onSubmit={handleSubmit}
                className="theme-form login-Form"
              >
                <Form.Group>
                  <Form.Label>Email Address</Form.Label>
                  <Form.Control
                    type="email"
                    name="email"
                    value={values.email}
                    onChange={handleChange}
                    isValid={touched.email && !errors.email}
                    placeholder="Enter your Email Address"
                    onBlur={handleBlur}
                    isInvalid={!!errors.email}
                  />
                  {touched.email && errors.email && (
                    <Form.Control.Feedback type="invalid">
                      {errors.email}
                    </Form.Control.Feedback>
                  )}
                </Form.Group>
                <Form.Group>
                  <Form.Label> Password</Form.Label>
                  <Form.Control
                    name="password"
                    type="password"
                    placeholder="Your Password"
                    value={values.password}
                    onChange={handleChange}
                    isValid={touched.password && !errors.password}
                    isInvalid={!!errors.password}
                    onBlur={handleBlur}
                  />
                  {touched.password && errors.password && (
                    <Form.Control.Feedback type="invalid">
                      {errors.password}
                    </Form.Control.Feedback>
                  )}
                </Form.Group>
                <ButtonSpinner
                  isSubmitting={isSubmitting}
                  title={"Sign in and Continue"}
                />
                {/* <Button
                  disabled={isSubmitting}
                  block
                  type="submit"
                  className="bt btn-secondary btn-secondary-big"
                >
                  {isSubmitting ? (
                    <Spinner animation="border" />
                  ) : (
                    "Sign in and Continue"
                  )}
                </Button> */}
              </Form>
            )}
          </Formik>
          <Col className="forget-signup-button p-0">
            <NavLink
              to={{ pathname: `${ROUTES.SIGNUP}` }}
              onClick={this.props.close}
              className="signup-button-login"
            >
              Sign up for free!
            </NavLink>

            <NavLink
              to={{ pathname: `${ROUTES.FORGET_PASSWORD}` }}
              onClick={this.props.close}
              className="forgetpassword-button"
            >
              Forgot Password?
            </NavLink>
          </Col>
        </Modal.Body>
      </Modal>
    );
  }
}

const mapStateToProps = (state) => ({
  isAuthenticated: state.auth.isAuthenticated,
  status: state.auth.status,
});

const mapDispatchToProps = {
  loginAction,
  getCurrentUser,
  getActiveAccountSideBar,
};

Login.propTypes = {
  history: PropTypes.object,
  close: PropTypes.func,
  getCurrentUser: PropTypes.func,
  loginApi: PropTypes.func,
  show: PropTypes.bool,
  loginAction: PropTypes.func,
  getActiveAccountSideBar: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Login));
