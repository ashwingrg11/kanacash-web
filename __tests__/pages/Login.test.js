import React from "react";
import Login from "../../src/pages/Login/Login";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import {shallow} from 'enzyme'
import store from "../../src/store/Store";

import renderer from "react-test-renderer";
describe("<Login />", () => {
  it("Login renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Login />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
   });

  //  it('Login Modal Component', () => {
  //   const wrapper = shallow( <Provider store={store}>
  //     <Router>  <Login />
  //       </Router>
  //     </Provider>);
  //   expect(wrapper.find('Modal').exists()).toEqual(true);
  //   expect(wrapper.find('div').exists()).toEqual(true);
  // });

});
