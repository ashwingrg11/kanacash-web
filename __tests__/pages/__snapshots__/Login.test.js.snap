// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Login renders correctly 1`] = `null`;
